ch_backup/logic/access.py

```diff
@@ -276,16 +282,60 @@ def _restore_replicated(
                 meta_data = acl_meta[str(i)]
                 name, obj_char = meta_data["name"], meta_data["char"]
 
-                # restore object data
-                file_path = os.path.join(restore_tmp_path, f"{uuid}.sql")
-                with open(file_path, "r", encoding="utf-8") as file:
-                    data = file.read()
-                    uuid_zk_path = _get_access_zk_path(context, f"/uuid/{uuid}")
-                    _zk_upsert_data(zk_client, uuid_zk_path, data)
+                uuid_zk_path = _get_access_zk_path(context, f"/uuid/{uuid}")
+                name_zk_path = _get_access_zk_path(
+                    context, f"/{obj_char}/{escape_metadata_file_name(name)}"
+                )
+                logging.debug(
+                    f'Upserting zk access entity uuid:"{uuid_zk_path}" name:"{name_zk_path}"'
+                )
 
-                # restore object link
-                uuid_zk_path = _get_access_zk_path(context, f"/{obj_char}/{name}")
-                _zk_upsert_data(zk_client, uuid_zk_path, uuid)
+                tx_insert = zk_client.transaction()
+                # Restore entity data
+                entiry_file_path = Path(restore_tmp_path) / f"{uuid}.sql"
+                entity_data = entiry_file_path.read_bytes()
+                tx_insert.create(uuid_zk_path, entity_data)
+                # Restore entity name
+                tx_insert.create(name_zk_path, uuid.encode())
+                result = tx_insert.commit()
+
+                if not any(isinstance(res, ZookeeperError) for res in result):
+                    # Transaction was completed successfully
+                    continue
+
+                # Handle conflicts
+                tx_update = zk_client.transaction()
+
+                # UUID conflict
+                if isinstance(result[0], NodeExistsError):
+                    tx_update.set_data(uuid_zk_path, entity_data)
+                else:
+                    tx_update.create(uuid_zk_path, entity_data)
+
+                # Name conflict
+                if isinstance(result[1], NodeExistsError):
+                    existing_uuid, _ = zk_client.get(name_zk_path)
+                    if uuid != existing_uuid:
+                        logging.debug(
+                            f"Remove conflicting UUID {existing_uuid.decode()}"
```

Comment:
We already are using non-transaction request, so will be critical to replace `if isinstance` -> `zk.exist(...)`?
If no, we can remove first transaction and simplify logic.